File: emneplanleggeren/src/model/Group.ts
import { GroupLecture } from "./GroupLecture";
import { Lecture } from "./Lecture";

class Group {
  readonly name: string;
  readonly groupLectures: Array<Lecture>;
  constructor(name: string, groupLectures: Array<GroupLecture> = []) {
    this.name = name;
    this.groupLectures = groupLectures;
  }
}
export { Group };

File: scraperScripts/group.py
from groupLecture import groupLecture
import json

class group():
    def __init__(self, name):
        self.name = name
        self.groupLectures = []

    def addLecture(self, groupLecture):
        self.groupLectures.append(groupLecture)

    def getNumber(self):
        return self.number
    
    def jsonExport(self):
        for lect in self.groupLectures:
            lect = lect.jsonExport()
        return json.dumps(
            self,
            default=lambda o: o.__dict__, 
            ensure_ascii=False
        )